packages/opentelemetry-exporter-collector/src/platform/browser/sendSpans.ts

```diff
@@ -0,0 +1,148 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { hrTime, hrTimeToTimeStamp } from '@opentelemetry/core';
+import { Logger } from '@opentelemetry/types';
+import { CollectorExporter } from '../../CollectorExporter';
+import * as collectorTypes from '../../types';
+
+/**
+ * function that is called once when {@link ExporterCollector} is initialised
+ * @param shutdownF shutdown method of {@link ExporterCollector}
+ */
+export function onInit(shutdownF: EventListener) {
+  window.addEventListener('unload', shutdownF);
+}
+
+/**
+ * function to be called once when {@link ExporterCollector} is shutdown
+ * @param shutdownF - shutdown method of {@link ExporterCollector}
+ */
+export function onShutdown(shutdownF: EventListener) {
+  window.removeEventListener('unload', shutdownF);
+}
+
+/**
+ * function to send spans to the [opentelemetry collector]{@link https://github.com/open-telemetry/opentelemetry-collector}
+ *     using the standard http/https node module
+ * @param spans
+ * @param onSuccess
+ * @param onError
+ * @param collectorExporter
+ */
+export function sendSpans(
+  spans: collectorTypes.Span[],
+  onSuccess: () => void,
+  onError: (status?: number) => void,
+  collectorExporter: CollectorExporter
+) {
+  const exportTraceServiceRequest: collectorTypes.ExportTraceServiceRequest = {
+    node: {
+      identifier: {
+        hostName: collectorExporter.hostName || window.location.host,
+        startTimestamp: hrTimeToTimeStamp(hrTime()),
+      },
+      libraryInfo: {
+        language: collectorTypes.LibraryInfoLanguage.WEB_JS,
+        // coreLibraryVersion: , not implemented
+        // exporterVersion: , not implemented
+        // coreLibraryVersion: , not implemented
+      },
+      serviceInfo: {
+        name: collectorExporter.serviceName,
+      },
+      // attributes: {}
```

Comment:
What would you think about making `attributes` and `resource` be optional parameters to the exporter config? Or maybe track that in a follow-up issue?

Reply:
not sure where would I get that information from, do we already have such information or it is not implemented yet ?

Reply:
I would see it just being called in like `serviceName` into the exporter, so it would be a matter of piping it through.

Reply:
I have piped attributes, but `resource` is not part of sdk yet